packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -60,8 +62,11 @@ Future<Depfile> copyAssets(Environment environment, Directory outputDirectory) a
         file.parent.createSync(recursive: true);
         final DevFSContent content = entry.value;
         if (content is DevFSFileContent && content.file is File) {
-          inputs.add(globals.fs.file(content.file.path));
-          if (!await iconTreeShaker.subsetFont(
+          inputs.add(content.file as File);
+          // Only tree shake MaterialIcons-Regular.ttf font.
+          final bool isMaterialIcons = globals.fs.path
+            .basename(content.file.path) == kMaterialIconsFontName;
+          if (!isMaterialIcons || !await iconTreeShaker.subsetFont(
```

Comment:
@dnfield fyi

Reply:
But... Why?  What about Cupertino icons or font awesome icins?

Reply:
The intention of this routine was to only do work if the font file was used by an IconData instance.  Is that not happening?

Reply:
As written this was spinning up a const finder instance for every single asset. Does this actually work with arbitrary icon sets? If so we could use something like the mime package to limit it to font files (ttf only?)

Reply:
The other change this would need is to limit the number of instances to 2 or 3, because each one seems to be quite heavy

Reply:
Ahh ok.  Yes, this should work with any font that you refernece a glyph in using `IconData` .. or at least any TTF.

Limiting this to 2 or 3 instances should be fine. And it wasn't meant to run on non-ttf assets.

Reply:
The expectation is that this code https://github.com/flutter/flutter/blob/3f30b6138d4155d91eb8899cc379b6726d4f2fa1/packages/flutter_tools/lib/src/build_system/targets/icon_tree_shaker.dart#L166 would return quickly, after the initial run of the const finder here: https://github.com/flutter/flutter/blob/3f30b6138d4155d91eb8899cc379b6726d4f2fa1/packages/flutter_tools/lib/src/build_system/targets/icon_tree_shaker.dart#L161

Why is the const finder running more than once? :\

Reply:
I figured it out, getIconData is fairly expensive, and we might be copying multiple assets at the same time. each one that hits this line before the first instance has finished will spawn a new const finder

Reply:
Ahh, yes.  Ok.  We should just trigger that to happen before the loop.  I'm still a little confused about how multiple places are reaching this code before the first one finishes -are we missing an await somewhere?

Reply:
Because the out loop is a Future.wait, not a for loop